webpush/encrypt.go

```diff
@@ -55,13 +55,43 @@ import (
 	"strings"
 )
 
+// ContentEncoding indicates the version of encoding.
+type ContentEncoding int
+
 const (
-	maxPayloadLength = 4078
+	// The encoding that is widely deployed with WebPush (as of 2016-11).
+	AESGCM ContentEncoding = iota
+	// The most recent encoding, the salt, record size and key identifier
+	// are included in a header that is part of the encrypted content coding.
+	AES128GCM
```

Comment:
specs refer to them as aesgcm and aes128gcm so I'm leaning towards keeping it, but also don't feel strongly about it. So if you do feel strongly I can change it in the next round.